### PR TITLE
chore(eslint): Config to run on JS files only

### PR DESCRIPTION
**Pull request by amithm7**
`eslint` throws error when fed non-js files.

### CODE REVIEW

**Review by ignoramous — COMMENTED** [review]
Doh. I never published my only comment...

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -57,12 +57,14 @@ jobs:
...
           git checkout ${{ env.PR_HEAD_SHA }}
 
-      - name: 🔧 Run eslint on changed files only
+      - name: 🔧 Run eslint on changed JS files only
         run: |
           DIFF=$(git diff --name-only --diff-filter=ACMRT \
             ${{ env.BASE_SHA }}...${{ env.PR_HEAD_SHA }})
 
-          npx eslint $DIFF --cache --fix
+          DIFF_JS=$(echo $DIFF | grep -E "\.(m|c)?js$")
```
[the remark this answers is not in the document]
Comment (amithm7) [reply .github/workflows/pr.yml:65]: @ignoramous , yeah

negated ignore pattern could be added to `eslintrc`.

https://github.com/serverless-dns/serverless-dns/blob/86a074b0bb3ec52cf1c5e95e9c8592e41ca7a4aa/.eslintrc.cjs#L20

Like this:

```js
module.exports = {
  // ...
  ignorePatterns: [
    "node_modules/",
    "dist/",
    "worker/",
    "test/data/cache/",

    // Ignore all files, except JS files
    "**/*.*",
    "!**/*.js",
    "!**/*.mjs",
    "!**/*.cjs",
  ],
  // ...
}
```

Would this be better?

[the remark this answers is not in the document]
Comment (ignoramous) [reply .github/workflows/pr.yml:65]: looks good to my unsuspecting eyes.

Btw, you mean:

```diff
- "!**/*.js",
+ "!**/*.ts",
```

right?

[the remark this answers is not in the document]
Comment (amithm7) [reply .github/workflows/pr.yml:65]: > Btw, you mean:
> 
> ```diff
> - "!**/*.js",
> + "!**/*.ts",
> ```
> right?

No,
`!**.*.js` for selecting js files.

eslint only works for JS (without [additional configuration](https://typescript-eslint.io/) that is)

**Comment by amithm7** [comment]
I wonder how "push to fork" worked. https://github.com/serverless-dns/serverless-dns/actions/runs/3728371589/jobs/6323321075

I thought it was broken...

File: .github/workflows/pr.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ignoramous) [reply .github/workflows/pr.yml]: Gotcha. Missed the `!`

**Comment by ignoramous** [comment]
> I thought it was broken...

🤷 GitHub action works in mysterious ways. It is free for FOSS projects, so I don't complain ;)

**Pull request merged**